test-driver/src/ComparisonContext.ts

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright 2012 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {IonEventStream} from "ion-js";
+import {Reader} from "ion-js";
+import {IonTypes, makeReader, Writer} from "ion-js";
+import fs from "fs";
+import {IonCompareArgs} from "./CliCompareArgs";
+import {ErrorType, IonCliError} from "./CliError";
+import {IonEvent, IonEventFactory, IonEventType} from "ion-js";
+
+/**
+ * ComparisonContext to create a structure for lhs and rhs streams
```

Comment:
Here and elsewhere: for classes whose structure/purpose is part of a spec, I think it would be helpful to provide a link to the full definition. That makes it less necessary to fully explain everything (including phrases like "lhs" and "rhs") in the class-level comments. 

I found [this](https://github.com/amzn/ion-test-driver#comparisonreport), but maybe there's something even better to point to?